cow_instrument/ComponentInfo.h

```diff
@@ -40,15 +42,26 @@ template <typename InstTree> class ComponentInfo {
               const double &theta, const Eigen::Vector3d &center);
 
 private:
+  /// initalization
+  void init();
+  /// Make maps with component index as key.
+  void makeInvertedMaps();
   /// Detector info
   std::shared_ptr<DetectorInfo<InstTree>> m_detectorInfo;
   /// Instrument tree
   const InstTree &m_instrumentTree;
   /// Inverted map to get detector indexes from component indexes
   std::vector<int64_t> m_componentToDetectorIndex;
-  /// Inverted map to get path component indexes indexes from component indexes
+  /// Inverted map to get path component indexes  from component indexes
   std::vector<int64_t> m_componentToPathIndex;
-  void makeInvertedMaps();
+  /// Inverted map to get branch node indexes from component indexes
+  std::vector<int64_t> m_componentToBranchNodeIndex;
+  /// Locally (branch node) indexed positions
```

Comment:
Note that there may also be components that are not detectors or path components, but not branch nodes either. Maybe a slit or a webcam. So using the term "branch node" everywhere below will be misleading. No need to change this now for the prototype though.